common/src/main/java/org/apache/uniffle/common/netty/protocol/SendShuffleDataRequest.java

```diff
@@ -95,15 +115,46 @@ public void encode(ByteBuf buf) {
     buf.writeLong(timestamp);
   }
 
-  private static Map<Integer, List<ShuffleBlockInfo>> decodePartitionData(ByteBuf byteBuf) {
-    Map<Integer, List<ShuffleBlockInfo>> partitionToBlocks = Maps.newHashMap();
+  public void decodeShuffleData(ByteBuf byteBuf) {
+    this.appId = ByteBufUtils.readLengthAndString(byteBuf);
+    encodedLength += ByteBufUtils.encodedLength(appId);
+    this.shuffleId = byteBuf.readInt();
+    encodedLength += Integer.BYTES;
+    encodedLength += Integer.BYTES; // stageAttemptNumber not in use, keep same with encodedLength.
+    this.requireId = byteBuf.readLong();
+    encodedLength += Long.BYTES;
+    this.partitionToBlocksInServer = decodePartitionData(byteBuf);
+    this.timestamp = byteBuf.readLong();
+    encodedLength += Long.BYTES;
+  }
+
+  public int getEncodedLength() {
+    return encodedLength;
```

Comment:
This make me confusing, what is the different between `getEncodedLength()` and `encodedLength`?